korge/src/commonMain/kotlin/com/soywiz/korge/view/Circle.kt

```diff
@@ -8,37 +9,39 @@ import com.soywiz.korma.geom.vector.*
  * The [autoScaling] determines if the underlying texture will be updated when the hierarchy is scaled.
  * The [callback] allows to configure the [Circle] instance.
  */
-inline fun Container.circle(radius: Double = 16.0, color: RGBA = Colors.WHITE, autoScaling: Boolean = true, callback: Circle.() -> Unit = {}): Circle = Circle(radius, color, autoScaling).addTo(this).apply(callback)
+inline fun Container.circle(
+    radius: Double = 16.0,
+    color: RGBA = Colors.WHITE,
+    autoScaling: Boolean = true,
+    callback: Circle.() -> Unit = {}
+): Circle = Circle(radius, color, autoScaling).addTo(this).apply(callback)
 
 /**
  * A [Graphics] class that automatically keeps a circle shape with [radius] and [color].
  * The [autoScaling] property determines if the underlying texture will be updated when the hierarchy is scaled.
  */
-open class Circle(radius: Double = 16.0, color: RGBA = Colors.WHITE, autoScaling: Boolean = true) : Graphics(autoScaling = autoScaling) {
+open class Circle(
+    radius: Double = 16.0,
+    color: RGBA = Colors.WHITE,
+    autoScaling: Boolean = true
+) : Graphics(autoScaling = autoScaling) {
+
     /** Radius of the circle */
-    var radius: Double = radius
-        set(value) {
-            field = value; updateGraphics()
-        }
+    var radius: Double by uiObservable(radius) { updateGraphics() }
+    /** Color of the circle */
+    var color: RGBA by uiObservable(color) { updateGraphics() }
 
     override val bwidth get() = radius * 2
     override val bheight get() = radius * 2
 
-    /** Color of the circle */
-    // @TODO: We could paint a white circle, and then tint it using the [colorMul] property
```

Comment:
@soywiz, since `Circle` view is still a `Graphics` and a `Container`, someone can add other views inside it. If we use `colorMul` to change the color of the circle, I suppose, we will add incorrect behavior for that case (`colorMul` will be applied to other views inside `Circle`).

Reply:
Makes sense.

https://github.com/korlibs/korge/pull/131/commits/39f2d77209cf70110a3839c3a488d9e7efee0ebc